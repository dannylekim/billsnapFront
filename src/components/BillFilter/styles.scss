@import "../../variables";

div.bill__filter__container {
  top: 75px;
  height: 220px;
  width: 500px;
  border-radius: 6px;
  background-color: $light-gray-background;

  div.filter__header {
    text-align: left;
    margin-left: 10px;

    font-family: $primary-font;
    font-size: 25px;

    span.filter__header__title {
      margin-left: 10px;
      position: relative;
      top: 2px;
      font-weight: bold;
    }
  }

  div.filter__selections__container {
    margin-top: 10px;
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    height: 120px;
    width: 90%;
    margin-left: auto;
    margin-right: auto;

    span.badge.badge-outline-primary.badge-pill {
      border: 1px solid $secondary-text-color;
      box-sizing: border-box;
      border-radius: 50px;
      color: $secondary-text-color;
      font-family: $primary-font;
      font-style: normal;
      font-weight: normal;
      font-size: 20px;
      line-height: 16px;
      width: 120px;
      margin-bottom: 10px;

      &.active {
        background-color: $secondary-color;
      }
    }

    span.badge.badge-outline-primary.badge-pill:hover {
      background-color: #bebebe;
    }
  }

  div.advanced__filter__selection {
    position: relative;
    bottom: 80px;
    left: 380px;
    width: 180px;
    height: 110px;
    font-family: $primary-font;

    label.custom-control.custom-checkbox {
      padding: 0;
      font-style: normal;
      font-weight: normal;
      font-size: 15px;
      color: $secondary-text-color;
    }

    input.form-control {
      box-sizing: border-box;
      color: rgba(0, 0, 0, 0.42);
      font-size: 15px;
      background: $card-color;
    }

    div.date__filter {
      span.custom-control-description {
        margin-left: 25px;
      }

      input.form-control {
        width: 172px;
        border: 3px solid rgba(0, 0, 0, 0.17);
        background: $card-color;
      }
    }

    div.category__filter {
      input.form-control {
        margin-top: 10px;
        width: 150px;
        border: 1px solid #becad6;
        background: $card-color;
      }
    }
  }
}
