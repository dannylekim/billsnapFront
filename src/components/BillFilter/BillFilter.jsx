import React, { useRef } from "react";
import { Badge, FormCheckbox, FormInput } from "shards-react";

import { useOutsideAlerter } from "../../helpers/ClickEvent";
import "./styles.scss";

/**
 * @description the filter box when three horizontal lines clicked
 */
const BillFilter = ({
  activeTab,
  applyFiltering,
  currentSorting,
  billStatusFilter,
  closeHandler,
  dateCheckboxHandler,
  dateFilters,
  filterToggles,
  filterToggleChange,
  handleDateSelection,
  handleStatusChange,
}) => {
  const wrapperRef = useRef(null);
  useOutsideAlerter(wrapperRef, closeHandler);

  const filterBadges = [
    {
      id: "descending",
      title: "Newest",
      onClick: () => applyFiltering("Newest"),
    },
    {
      id: "ascending",
      title: "Oldest",
      onClick: () => applyFiltering("Oldest"),
    },
    {
      id: "by__status",
      title: "Status",
      onClick: () =>
        filterToggleChange({
          statusOpened: true,
          categoryOpened: false,
          dateOpened: false,
        }),
    },
    {
      id: "alphabetical__increasing",
      title: "A to Z",
      onClick: () => applyFiltering("A to Z"),
    },
    {
      id: "alphabetical__decreasing",
      title: "Z to A",
      onClick: () => applyFiltering("Z to A"),
    },
    {
      id: "by__category",
      title: "Category",
      onClick: () =>
        filterToggleChange({
          categoryOpened: true,
          statusOpened: false,
          dateOpened: false,
        }),
    },
    {
      id: "by__date",
      title: "Date",
      onClick: () =>
        filterToggleChange({
          dateOpened: true,
          categoryOpened: false,
          statusOpened: false,
        }),
    },
  ];

  const extraFilterSelected =
    filterToggles.dateOpened === true ||
    filterToggles.statusOpened === true ||
    filterToggles.categoryOpened === true;
  
  return (
    <div className='bill__filter__container' ref={wrapperRef}>
      <div className='filter__header'>
  <span className='filter__header__title'>Filters {activeTab}</span>
      </div>
      <div className='filter__selections__container'>
        {filterBadges.map((badgeObject, key) => (
          <Badge
            key={key}
            className={badgeObject.title === currentSorting && 'active'}
            outline
            pill
            id={badgeObject.id}
            onClick={badgeObject.onClick}>
            {badgeObject.title}
          </Badge>
        ))}
      </div>
      {extraFilterSelected && (
        <div className='advanced__filter__selection'>
          {filterToggles.dateOpened === true && (
            <div className='date__filter'>
              <FormCheckbox
                checked={dateFilters.startDate.selected}
                onChange={() => dateCheckboxHandler("startDate")}>
                Start date
                {dateFilters.startDate.value !== ""
                  ? `: ${dateFilters.startDate.value}`
                  : ""}
              </FormCheckbox>
              <FormCheckbox
                checked={dateFilters.endDate.selected}
                onChange={() => dateCheckboxHandler("endDate")}>
                End date
                {dateFilters.endDate.value !== ""
                  ? `: ${dateFilters.endDate.value}`
                  : ""}
              </FormCheckbox>
              <FormInput
                type='date'
                onChange={(event) => handleDateSelection(event)}
                disabled={
                  dateFilters.endDate.selected === false &&
                  dateFilters.startDate.selected === false
                }
              />
            </div>
          )}
          {filterToggles.statusOpened === true && (
            <div className='status__filter'>
              <FormCheckbox
                checked={billStatusFilter.resolved}
                onChange={() =>
                  handleStatusChange('resolved')
                }>
                PAID
              </FormCheckbox>
              <FormCheckbox
                checked={billStatusFilter.open}
                onChange={() =>
                  handleStatusChange('open')
                }>
                UNPAID
              </FormCheckbox>
              <FormCheckbox
                checked={billStatusFilter.in_progess}
                onChange={() =>
                  handleStatusChange('in_progess')
                }>
                ONGOING
              </FormCheckbox>
            </div>
          )}
          {filterToggles.categoryOpened === true && (
            <div className='category__filter'>
              <FormInput type='text' placeholder='Restaurant' />
            </div>
          )}
        </div>
      )}
    </div>
  );
};

export default BillFilter;
