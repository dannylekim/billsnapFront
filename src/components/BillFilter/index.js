import BillFilter from "./BillFilter.jsx";

export default BillFilter;
