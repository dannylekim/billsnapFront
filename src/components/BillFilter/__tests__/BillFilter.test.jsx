import React from "react";
import BillFilter from "../BillFilter";
import { shallow } from "enzyme";

describe("BillFilter", () => {
    let wrapper;
    describe("render", () => {
        describe("snapshots ðŸ“¸", () => {
            it("BillFilter should match snap shot when category filter opened", () => {
                matches(
                <BillFilter dateFilters={{
                                            startDate: {selected: false, value: ""},
                                            endDate: {selected: false, value: ""}
                                        }}
                            filter={{ opened: true, type: "", statusOpened: false, categoryOpened: true, dateOpened: false }} 
                            billStatusFilter={{resolved: false, open: false, in_progess:false}}
                            handleDateSelection={jest.fn()}
                            updateBills={jest.fn()}
                            setState={jest.fn()} />
                );
            });
        });

       
    });

        describe("component ", () => {
            it("should show the date value next to start date", () => {
                wrapper = shallow( <BillFilter dateFilters={{
                                                    startDate: {selected: true, value: "2020-01-01"},
                                                    endDate: {selected: false, value: ""}
                                                }}
                                    filter={{ opened: true, type: "", statusOpened: false, categoryOpened: false, dateOpened: true }} 
                                    billStatusFilter={{resolved: false, open: false, in_progess:false}}
                                    handleDateSelection={jest.fn()}
                                    updateBills={jest.fn()}
                                    setState={jest.fn()} />);

                expect(wrapper.find("FormCheckbox").at(0).render().text()).toBe("Start date : 2020-01-01");

            })

            it("should show the date value nex to end date", () => {
                wrapper = shallow( <BillFilter dateFilters={{
                                                    startDate: {selected: false, value: "" },
                                                    endDate: {selected: true, value: "2020-01-01"}
                                                }}
                                                filter={{ opened: true, type: "", statusOpened: false, categoryOpened: false, dateOpened: true }} 
                                                billStatusFilter={{resolved: false, open: false, in_progess:false}}
                                                handleDateSelection={jest.fn()}
                                                updateBills={jest.fn()}
                                                setState={jest.fn()} />);

                expect(wrapper.find("FormCheckbox").at(1).render().text()).toBe("End date : 2020-01-01");

            })
        })

})

