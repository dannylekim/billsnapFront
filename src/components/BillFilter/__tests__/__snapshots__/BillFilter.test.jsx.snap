// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillFilter render snapshots ðŸ“¸ BillFilter should match snap shot when category filter opened 1`] = `
<div
  className="bill__filter__container"
>
  <div
    className="filter__header"
    onClick={[Function]}
  >
     
    <svg
      fill="currentColor"
      height={24}
      size={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      />
    </svg>
    <span
      className="filter__header__title"
    >
      Filters
    </span>
  </div>
  <div
    className="filter__selections__container"
  >
    <span
      className="badge badge-outline-primary badge-pill"
      id="descending"
      onClick={[Function]}
    >
      Newest
    </span>
    <span
      className="badge badge-outline-primary badge-pill"
      id="ascending"
      onClick={[Function]}
    >
      Oldest
    </span>
    <span
      className="badge badge-outline-primary badge-pill"
      id="by__status"
      onClick={[Function]}
    >
      Status
    </span>
    <span
      className="badge badge-outline-primary badge-pill"
      id="alphabetical__increasing"
      onClick={[Function]}
    >
      A to Z
    </span>
    <span
      className="badge badge-outline-primary badge-pill"
      id="alphabetical__decreasing"
      onClick={[Function]}
    >
      Z to A
    </span>
    <span
      className="badge badge-outline-primary badge-pill"
      id="by__category"
      onClick={[Function]}
    >
      Category
    </span>
    <span
      className="badge badge-outline-primary badge-pill"
      id="by__date"
      onClick={[Function]}
    >
      Date
    </span>
  </div>
  <div
    className="advanced__filter__selection"
  >
    <div
      className="category__filter"
    >
      <input
        className="form-control"
        placeholder="Restaurant"
        type="text"
      />
    </div>
  </div>
</div>
`;
